### PR TITLE
Rearrangements

### PR DESCRIPTION
**Pull request by markkramerus**
Moved some parts around for clarity.

### CODE REVIEW

**Comment by cmoesel** [comment]
It's hard to tell what really changed here since it all moved around.  I see some real edits in the top-sections, but should I assume that once we get to line 183 it's mainly just moving sections around?

File: input/pagecontent/reference.md [hunk 1]
```diff
@@ -668,12 +551,12 @@ Assignment rules follow this syntax:
 * <element> = {value}
 ```
 
-The left side of this expression follows the [FSH path grammar](#fsh-paths). The data type on the right side must align with the data type of the final element in the path, and may be a complex data type (such as an address).
+The left side of this expression follows the [FSH path grammar](#fsh-paths). The data type on the right side must align with the data type of the final element in the path, and may be a primitive or complex data type or a resource reference.
```
Comment (ngfreiter) [inline input/pagecontent/reference.md:554]: I think this is a bit confusingly worded. What do you mean by resource reference here? Do you mean the `Reference` keyword, or just the name of an Instance.

**Comment by ngfreiter** [comment]
@cmoesel, it doesn't seem to be purely stuff moving around. There are some additions. For example, some of the content at the end in the "Assignments involving Instances" section looked new to me.

**Comment by ngfreiter** [comment]
If the date on the bottom right of both sides is meant to reflect when it was last updated, we should change it to July 2020.

**Comment by ngfreiter** [comment]
In the syntax section on Alias, there should be a "_$_" in front of the alias name, since you can optionally add that.

**Comment by ngfreiter** [comment]
On the Instance example, you have 
```
Usage: Example
```
which should be:
```
Usage: #example
```

**Comment by ngfreiter** [comment]
On the Profile example, why does the `Parent` have `$USCorePatient`? I guess because you are using an Alias, but this seems strange to define an Alias, when you could just reference the profile by name, `USCorePatientProfile`, and generally I think that would be the preferred/simpler way of doing things.

**Comment by ngfreiter** [comment]
And in fact on the mapping, you just say `USCorePatient`, so at least those should match.

File: input/pagecontent/reference.md [hunk 2]
```diff
@@ -178,56 +180,31 @@ These comments can take up multiple lines.
...
-* For all other non-whitespace lines, detect the smallest number of leading spaces and trim that from the beginning of every line.
-
-#### Coded Data Types
+<pre><code>{CodeSystem name|id|url}#{code} <i>"{display string}"</i>
 
-The four "coded" types in FHIR are code, Coding, CodeableConcept, and Quantity. FSH provides special grammar for expressing codes and assigning coded values for these data types.
+{CodeSystem name|id|url}|{version string}#{code} <i>"{display string}"</i></code></pre>
 
-##### Representing the code Data Type
+As [indicated by italics](#about-this-guide), the `"{display string}"` is optional. `CodeSystem` represents the controlled terminology that the code is taken from. The bar syntax for the version of the code system is the same approach used in the canonical data type in FHIR. An alternative to the bar syntax is to set the version element of Coding directly (see examples). To set the less-common properties of a Coding, [assignment rules](#assignments-with-the-coding-data-type) can be used.
```
[the remark this answers is not in the document]
Comment (markkramerus) [reply input/pagecontent/reference.md:206]: That sentence didn't belong anymore, with the rearrangements. I cut it.

File: input/pagecontent/reference.md (hunk 1 above)
Comment (markkramerus) [reply input/pagecontent/reference.md:555]: Fixed.

**Comment by markkramerus** [comment]
> @cmoesel, it doesn't seem to be purely stuff moving around. There are some additions. For example, some of the content at the end in the "Assignments Involving Instances" section looked new to me.

I reorganized the Assignments section into subsections.

**Comment by markkramerus** [comment]
> On the Examples side, the second extension example:
> 
> ```
> extension[http://hl7.org/fhir/StructureDefinition/location-distance]
> ```
> 
> seems a bit confusing. I think someone would only really want to do this if they were adding this extension to an Instance when it wasn't already on the profile, which is a fine thing to do, but it would probably be best practice to use an Alias there, and it seems to me like too advanced a use case for a quick reference. Might be confusing to have without added explanation.

I am going to leave this, because it is a valid to refer to the extension by its URL, even after you give it a local name.

**Comment by markkramerus** [comment]
> On the Profile example, why does the `Parent` have `$USCorePatient`? I guess because you are using an Alias, but this seems strange to define an Alias, when you could just reference the profile by name, `USCorePatientProfile`, and generally I think that would be the preferred/simpler way of doing things.

Simplified that as you suggested.

**Pull request merged**